packages/odyssey-react/src/components/Modal/Modal.tsx

```diff
@@ -111,9 +122,14 @@ export const Modal = withTheme(
     const oid = useOid(id);
     const modalDialog = useRef<HTMLDivElement>(null);
     const componentClass = useCx(styles.root, { [styles.openState]: open });
-
-    if (open && onOpen) {
-      onOpen();
+    const { restoreFocus, setFocus } = useFocus();
+    const lastFocusedElemRef = useRef<OptionalHTMLElement>(null);
+
+    if (open) {
+      lastFocusedElemRef.current = setFocus(modalDialog.current);
+      onOpen && onOpen();
+    } else {
+      lastFocusedElemRef.current && restoreFocus(lastFocusedElemRef.current);
```

Comment:
the issue with shadowed props and state are no longer problematic? I can close the modal and re-open it now without the bug where the prop stays closed?

Reply:
Correct, it handles multiple open/close cycles since we're not managing the open/close state in the component.  Incidentally, given the Modal component doesn't manage open/closed state, it's quite difficult (if not impossible) to have the component internally manage the open/closed state, preventing us from implementing the "close on `Esc` key press" and "close when clicking outside the modal" behaviors without a significant amount of work.  Those efforts are tracked in a separate ticket and are outside the scope of this PR.

Reply:
@kamalgill-okta thanks these are all problems we can solve!